### PR TITLE
logout && doc update

### PR DESCRIPTION
**Pull request by knockknockyoo**
1. Updated logout API
- NIH & Login.gov accept logout API

2. ejs views template updated
- display IDP type & user's name
- NIH / Google logout

3. Added test routes for development purpose

4. Added restful API test codes

### CODE REVIEW

File: idps/index.js [hunk 1]
```diff
@@ -2,13 +2,14 @@ const googleClient = require('./google');
...
+const {NIH, GOOGLE, LOGIN_GOV} = require("../constants/idp-constants");
 
 const oauth2Client = {
     login: async (code, idp, redirectingURL) => {
         // if google
-        if (isCaseInsensitiveEqual(idp, 'google')) {
+        if (isCaseInsensitiveEqual(idp, GOOGLE)) {
             return googleClient.login(code, redirectingURL);
-        } else if (isCaseInsensitiveEqual(idp,'NIH')) {
+        } else if (isCaseInsensitiveEqual(idp,NIH) || isCaseInsensitiveEqual(idp, LOGIN_GOV)) {
```
Comment (knockknockyoo) [inline idps/index.js:12]: Login.gov and NIH will go through authentication.

**Review by knockknockyoo — COMMENTED** [review]
Login.gov also added to pass login authentication.

**Review by n2iw — CHANGES_REQUESTED** [review]
- README.md line 51 should be "If set to development, a test ...."
- There are unwanted code in the PR such as nihRouter 
- There is a bug related to detect NIH or Login.gov user, if a user use xxxx@nih.gov to register a login.gov user, when he/she logs in, the IDP detected is "NIH" instead of "login.gov"

File: README.md [hunk 2]
```diff
@@ -46,4 +46,6 @@ Following environmental variables are needed
 - NIH_TOKEN_URL: NIH API address to create token for login
 - NIH_LOGOUT_URL: NIH API address to invalidate token for logout
 - NIH_SCOPE: space-separated lists of identifiers to specify access privileges
-- NIH_PROMPT: 
\ No newline at end of file
+- NIH_PROMPT: to force re-authorization event when a current session is still active
+# Local development configuration
+- NODE_ENV: If set to "development", a test html page will be activated in the route "/"
```
Comment (knockknockyoo) [inline README.md:51]: updated NODE_ENV description

File: idps/nih.js [hunk 3]
```diff
@@ -1,10 +1,10 @@
-const {getNIHToken, nihUserInfo, nihLogout, getIDP, isNIHLogin} = require("../services/nih-auth");
+const {getNIHToken, nihUserInfo, nihLogout, getIDP} = require("../services/nih-auth");
 const client = {
     login: async (code, redirectingURL) => {
         const token = await getNIHToken(code, redirectingURL);
         const user = await nihUserInfo(token);
         // use a preferred name or email as identity
-        const idp = getIDP(user['preferred_username'] && !isNIHLogin(user.email) ? user['preferred_username'] : user.email);
+        const idp = getIDP(user['preferred_username'] ? user['preferred_username'] : user.email);
```
Comment (knockknockyoo) [inline idps/nih.js:7]: preferred_username will be used to determine as a login.gov authenticator

**Review by knockknockyoo — COMMENTED** [review]
1. deleted nih router
2. updated README.md
3. preferred_name is only indicator as a login.gov authenticator

**Review by n2iw — APPROVED** [review]

**Pull request merged**